deepcell/image_generators/cropping.py

```diff
@@ -0,0 +1,576 @@
+            else:
+                row_start = np.random.randint(0, img_dims[0] - self.crop_size[0])
+                col_start = np.random.randint(0, img_dims[1] - self.crop_size[1])
+                crop_indices = ([row_start, row_start + self.crop_size[0]],
+                                [col_start, col_start + self.crop_size[1]])
+
+        transform_parameters['crop_indices'] = crop_indices
+
+        return transform_parameters
+
+    def apply_transform(self, x, transform_parameters):
+
+        if transform_parameters['crop_indices'] is not None:
+            row_indices, col_indices = transform_parameters['crop_indices']
+            if self.channel_axis == 1:
+                x = x[:, row_indices[0]:row_indices[1], col_indices[0]:col_indices[1]]
+            else:
+                x = x[row_indices[0]:row_indices[1], col_indices[0]:col_indices[1], :]
+
+        x = ImageDataGenerator.apply_transform(self,
+                                               x=x,
+                                               transform_parameters=transform_parameters)
+        return x
+
+    def random_transform(self, x, y=None, seed=None):
+        """Applies a random transformation to an image.
+
+        Args:
+            x (numpy.array): 3D tensor or list of 3D tensors,
+                single image.
+            y (numpy.array): 3D tensor or list of 3D tensors,
+                label mask(s) for x, optional.
+            seed (int): Random seed.
+
+        Returns:
+            numpy.array: A randomly transformed copy of the input (same shape).
+                If y is passed, it is transformed if necessary and returned.
+        """
+        params = self.get_random_transform(x.shape, seed)
+
+        if isinstance(x, list):
+            x = [self.apply_transform(x_i, params) for x_i in x]
+        else:
+            x = self.apply_transform(x, params)
+
+        if y is None:
+            return x
+
+        # Nullify the transforms that don't affect `y`
+        params['brightness'] = None
+        params['channel_shift_intensity'] = None
+        _interpolation_order = self.interpolation_order
+        self.interpolation_order = 0
+
+        if isinstance(y, list):
+            y_new = []
+            for y_i in y:
+                if y_i.shape[self.channel_axis - 1] > 1:
+                    y_t = self.apply_transform(y_i, params)
+
+                # Keep original interpolation order if it is a
+                # regression task
+                elif y_i.shape[self.channel_axis - 1] == 1:
+                    self.interpolation_order = _interpolation_order
+                    y_t = self.apply_transform(y_i, params)
+                    self.interpolation_order = 0
+                y_new.append(y_t)
+            y = y_new
+        else:
+            y = self.apply_transform(y, params)
+
+        self.interpolation_order = _interpolation_order
+        return x, y
+
+    def fit(self, x, augment=False, rounds=1, seed=None):
```

Comment:
Is it necessary to override the parent class' version?

Reply:
Yes, lines 531-535 are new

Reply:
I think there is a test function we can add to the test suite for the `fit` method. Try looking at the MovieDataGenerator tests.

Reply:
I didn't see anything beyond just calling fit, which the code currently does